ap/columnar_util.py

```diff
@@ -863,7 +922,7 @@ def requires(self, func: Callable[[law.Task, dict], dict]) -> None:
         """
         self.requires_func = func
 
-    def run_requires(self, task: law.Task, reqs: Optional[dict] = None):
+    def run_requires(self, task: law.Task, reqs: Optional[dict] = None) -> dict:
```

Comment:
Also propagate to `self.shifts` (see comment for `run_update`)

Reply:
True! I'll address this in the next commit.